File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_cpp/code_output/BinaryDataProcessor.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <iomanip>
#include <sstream>

class BinaryInfo {
private:
    double zeroes;
    double ones;
    int bitLength;

public:
    BinaryInfo(double zeroes, double ones, int bitLength)
        : zeroes(zeroes), ones(ones), bitLength(bitLength) {}

    double getZeroes() const { return zeroes; }
    double getOnes() const { return ones; }
    int getBitLength() const { return bitLength; }

    std::string toString() const {
        std::ostringstream oss;
        oss << std::fixed << std::setprecision(3)
            << "{Zeroes: " << zeroes << ", Ones: " << ones
            << ", Bit length: " << bitLength << "}";
        return oss.str();
    }

    friend std::ostream& operator<<(std::ostream& os, const BinaryInfo& info) {
        os << info.toString();
        return os;
    }
};

class BinaryDataProcessor {
private:
    std::string binaryString;

public:
    BinaryDataProcessor(const std::string& binaryString)
        : binaryString(binaryString) {
        cleanNonBinaryChars();
    }

    void cleanNonBinaryChars() {
        binaryString.erase(
            std::remove_if(binaryString.begin(), binaryString.end(),
                [](char c) { return c != '0' && c != '1'; }),
            binaryString.end());
    }

    BinaryInfo calculateBinaryInfo() {
        int zeroesCount = static_cast<int>(std::count(binaryString.begin(), binaryString.end(), '0'));
        int onesCount = static_cast<int>(std::count(binaryString.begin(), binaryString.end(), '1'));
        int totalLength = static_cast<int>(binaryString.length());

        double zeroesPercentage = static_cast<double>(zeroesCount) / totalLength;
        double onesPercentage = static_cast<double>(onesCount) / totalLength;

        return BinaryInfo(zeroesPercentage, onesPercentage, totalLength);
    }

    std::string convertToAscii() {
        std::string asciiString;
        for (size_t i = 0; i < binaryString.length(); i += 8) {
            std::string byteString = binaryString.substr(i, 8);
            int decimal = std::stoi(byteString, nullptr, 2);
            asciiString += static_cast<char>(decimal);
        }
        return asciiString;
    }

    std::string convertToUtf8() {
        std::string utf8String;
        for (size_t i = 0; i < binaryString.length(); i += 8) {
            std::string byteString = binaryString.substr(i, 8);
            int decimal = std::stoi(byteString, nullptr, 2);
            utf8String += static_cast<char>(decimal);
        }
        return utf8String;
    }

    std::string getBinaryString() const {
        return binaryString;
    }
};

int main() {
    BinaryDataProcessor bdp("0110100001100101011011000110110001101111");
    std::cout << bdp.getBinaryString() << std::endl;
    std::cout << bdp.calculateBinaryInfo() << std::endl;
    std::cout << bdp.convertToAscii() << std::endl;
    std::cout << bdp.convertToUtf8() << std::endl;
    return 0;
}

File: data/derived_cot/rq3_prompting/results/translation/qwen/cpp_to_py/code_output/BinaryDataProcessor.py
class BinaryDataProcessor:
    def __init__(self, binary_string):
        self.binary_string = self.clean_non_binary_chars(binary_string)
    
    def clean_non_binary_chars(self, s):
        return ''.join(filter(lambda c: c in '01', s))
    
    def calculate_binary_info(self):
        total_length = len(self.binary_string)
        zeroes_count = self.binary_string.count('0')
        ones_count = total_length - zeroes_count
        
        zeroes_percentage = zeroes_count / total_length
        ones_percentage = ones_count / total_length
        
        return {
            "Zeroes": zeroes_percentage,
            "Ones": ones_percentage,
            "Bit length": total_length
        }
    
    def convert_to_ascii(self):
        s = []
        for i in range(0, len(self.binary_string), 8):
            chunk = self.binary_string[i:i+8]
            if chunk:
                num = 0
                for j in range(len(chunk)):
                    if chunk[j] == '1':
                        num |= (1 << (7 - j))
                s.append(chr(num))
        return ''.join(s)
    
    def convert_to_utf8(self):
        return self.convert_to_ascii()